### ISSUE TITLE
NOLOL label vs YOLOL optimized variable - collision

### ISSUE THREAD

**Issue by elewarr** [dbaumgarten/yodk#75]
Given script
```
include "std/math"

var1 = 0
var2 = 0

while :a do
    while :b > 100 do
        last_range = :b
    end

    :c = math_floor(last_range)
    inv = 1
    while :v >= :ma do
        :c -= 2 * inv
        inv *= -1
    end
end
e>
```
produces
```
b=0 c=0
ifnot :a thengoto1end d=0 e=1
if:b then:c*=4 e*=-1 goto3end goto2
goto1
```

Variable `:c` should be inverted, it's multiplied by `4` instead. If `e>` is replaced with another (non-conflicting) variable then the output is correct.

**Comment by dbaumgarten** [comment]
Hi, I could reproduce your issue, even though your provided output seems incorrect.
Compiling your nolol-code I get:
```
b=0 c=0
ifnot :a thengoto1end
if:b>100thend=:b goto3end :c=d-d%1 e=1
if:v>=:ma then:c-=10 e*=-1 goto4end goto2
goto1
```
which clearly shows the behavior you described.

The optimized variable-name "e" is treated as a jump-label and therefore the generated code is incorrect.

That should be fixable, but I am not sure what the best way would be. The simplest one would probably be to have a blacklist of varnames for the varname-optimizer and add all jump-labels to that blacklist. So in this specific case "inv" would instead be translated to "f".

I wonder if you found this by accident, or if you were actively trying to generate such a collision. I just never expected people to use single-character jump-labels

**Comment by elewarr** [comment]
> [...] even though your provided output seems incorrect.

Yes, I must have changed something before I pasted YOLOL.

> I wonder if you found this by accident [...]

I'm actively using your compiler (VSCode plugin) and writing scripts. I'd say half of my game time is actually copy-pasting scripts. I wish there'd be a better way.

`YODK` is amazing and big thanks from me!

I found it by accident. Had label `f` for "finish" and enough local variables to generate collision. The code didn't work as expected - I was surprised that "the optimization" wen't so far. Then I realized it was the label. 😄 

> I just never expected people to use single-character jump-labels

I thought it was the official recommendation. 😉 
https://dbaumgarten.github.io/yodk/#/nolol?id=line-labels

**Comment by dbaumgarten** [comment]
I already prepared a fix for this, but I am not 100% satisfied with it. Anyway, some fix will come with the next version.
Until then, just avoid one-char-labels (ignoring any other official recommendations ;) ).

By the way: At least for pasting scripts into Starbase, there is a better way then copying line by line: https://dbaumgarten.github.io/yodk/#/vscode-instructions?id=auto-typing-into-starbase

**Comment by elewarr** [comment]
> there is a better way then copying line by line

What kind of sorcery this is? Thank you 😲, my life just got a bit easier.

**Comment by dbaumgarten** [comment]
Fixed as part of v0.1.7
